python/hail/expr/types.py

```diff
@@ -1,10 +1,10 @@
 import abc
-from hail.java import jindexed_seq
-from hail.java import scala_object, Env, jset
-from hail.representation import GenomeReference
-from hail.representation import Variant, AltAllele, Locus, Interval, Struct, Call
-from hail.typecheck import *
+
 from hail.history import *
+from hail.typecheck import *
+from hail.utils import Struct
+from hail.utils.java import scala_object, jset, jindexed_seq, Env
+import hail.genetics as genetics
```

Comment:
So I feel like non-genetics types, expr, etc., shouldn't import the genetics module.  Also, should TVariant, etc. be in genetics?  Right now it's in types.  I'd almost imagine that genetics module structure should mirror the structure of the non-genetics module: types, instance objects, functions, methods on Matrix.

Reply:
This seems very reasonable. Could I save this for the next PR? I'd prefer to do as much as possible incrementally.

Reply:
I agree with Cotton on this.

Reply:
What I want to consider carefully is how this will look to a genetics user. If it's just `from hail import *; from hail.genetics import *` that seems OK. But what happens to `functions`? It's duplicated and now you need to do `from hail.expr import functions; from hail.genetics.expr import genetics_functions`?

Reply:
totally agree they should be separate, though.